Dockerfile

```diff
@@ -1,72 +1,48 @@
 RUN wget https://cmake.org/files/v3.7/cmake-3.7.2.tar.gz && tar -xvf cmake-3.7.2.tar.gz
 RUN cd cmake-3.7.2 && ./bootstrap && make && make install && cd /home
-RUN apt-get update && apt-get install -y automake
 
-# install libtool
-RUN apt-get update && apt-get install -y libtool
-
-# Install flex
-#RUN apt-get update && apt-get install -y flex
-#
-#install bison 2.7.1
-#RUN wget http://launchpadlibrarian.net/140087283/libbison-dev_2.7.1.dfsg-1_amd64.deb
-#RUN wget http://launchpadlibrarian.net/140087282/bison_2.7.1.dfsg-1_amd64.deb
-#RUN dpkg -i libbison-dev_2.7.1.dfsg-1_amd64.deb
-#RUN dpkg -i bison_2.7.1.dfsg-1_amd64.deb
-# To prevent update manager from overwriting this package
-#RUN apt-mark hold libbison-dev
-#RUN apt-mark hold bison
+# install libtool and automake
+RUN apt-get update && apt-get install -y automake libtool
 
 # Install librdkafka.
-ADD ./librdkafka /home/asn1_codec/librdkafka
-RUN cd /home/asn1_codec/librdkafka && ln -s /usr/bin/python3 /usr/bin/python && ./configure && make && make install
+ADD ./librdkafka /asn1-codec/librdkafka
+RUN cd /asn1-codec/librdkafka && ln -s /usr/bin/python3 /usr/bin/python && ./configure && make && make install
 
 # Install pugixml
-ADD ./pugixml /home/asn1_codec/pugixml
-RUN cd /home/asn1_codec/pugixml && mkdir -p build && cd build && cmake .. && make && make install
+ADD ./pugixml /asn1-codec/pugixml
+RUN cd /asn1-codec/pugixml && mkdir -p build && cd build && cmake .. && make && make install
 
 # Build and install asn1c submodule
-ADD ./asn1c /home/asn1_codec/asn1c
-RUN cd /home/asn1_codec/asn1c && test -f configure || autoreconf -iv && ./configure --disable-test-asan --disable-test-ubsan && make
-#RUN cd /home/asn1_codec/asn1c && make check
-RUN cd /home/asn1_codec/asn1c && make install
+ADD ./asn1c /asn1-codec/asn1c
+RUN cd /asn1-codec/asn1c && acloacl && test -f configure || autoreconf -iv && ./configure && make && make install
+
+# Generate ASN.1 API.
+RUN export LD_LIBRARY_PATH=/usr/local/lib
+ADD ./asn1c_combined /asn1-codec/asn1c_combined
+ADD ./scms-asn /asn1-codec/scms-asn
+RUN cd /asn1-codec/asn1c_combined && bash doIt.sh
 
-# echo Generating ASN.1 API ...
-ADD ./asn1c_combined /home/asn1_codec/asn1c_combined
-ADD ./scms-asn /home/asn1_codec/scms-asn
-RUN cd /home/asn1_codec/asn1c_combined && chmod +x ./doIt.sh && ./doIt.sh
+# add the source and build files
+ADD CMakeLists.txt /asn1-codec
+ADD ./config /asn1-codec/config
+ADD ./include /asn1-codec/include
+ADD ./src /asn1-codec/src
+ADD ./kafka-test /asn1-codec/kafka-test
+ADD ./unit-test-data /asn1-codec/unit-test-data
 
-# build asn1_codec
-ADD CMakeLists.txt /home/asn1_codec
-ADD ./config /home/asn1_codec/config
-ADD ./include /home/asn1_codec/include
-ADD ./src /home/asn1_codec/src
-RUN cd /home/asn1_codec && mkdir -p build && cd build && cmake .. && make
+# Build acm.
+RUN cd /asn1-codec && mkdir -p /build && cd /build && cmake /asn1-codec && make
 
-RUN echo "export LD_LIBRARY_PATH=/usr/local/lib" >> ~/.profile
```

Comment:
Please do not remove these statements as they are often needed when debugging issues from within the container. Keep in mind that running these applications inside docker container is now the primary mode of deployment, even for WYDOT.